05-java-mqtt/src/main/java/demo/data/MqttProtelisNode.java

```diff
@@ -0,0 +1,77 @@
+package demo.data;
+
+import java.beans.ConstructorProperties;
+import java.util.List;
+
+/**
+ * POJO class representing a Protelis node.
+ */
+public final class MqttProtelisNode {
+    private final int id;
+    private final String listen;
+    private final List<String> neighbors;
+    private final boolean leader;
+
+    /**
+     * Constructor method.
+     *
+     * @param id        the unique id of the node
+     * @param listen    the topic listened by the node
+     * @param neighbors the neighbors of the node
+     * @param leader    whether the node is a leader or not
+     */
+    @ConstructorProperties({"id", "listen", "neighbors", "leader"})
```

Comment:
I do not know this annotation: what does it do?

Reply:
It's used by Konf to map TOML object properties with the class attributes.

Reply:
Nice!